forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,376 @@
+            .map_err(handle_rpc_err)?;
+
+        let network = chain_get_name((), &config.client.rpc_token)
+            .await
+            .map_err(handle_rpc_err)?;
+
+        // Wallet info
+        let default_wallet_address = wallet_default_address((), &config.client.rpc_token)
+            .await
+            .map_err(handle_rpc_err)?;
+
+        let chain_head = chain_head(&config.client.rpc_token)
+            .await
+            .map_err(handle_rpc_err)
+            .context("couldn't fetch chain head, is the node running?")?;
+
+        let mut ts = chain_head.0.clone();
+        let mut tipsets = vec![];
+
+        for _ in 0..(config.chain.policy.chain_finality - 1).min(ts.epoch()) {
+            let parent_tipset_keys = TipsetKeysJson(ts.parents().clone());
+            let tsjson = chain_get_tipset((parent_tipset_keys,), &config.client.rpc_token)
+                .await
+                .map_err(handle_rpc_err)
+                .context("Failed to fetch tipset.")?;
+            tipsets.push(tsjson.0.clone());
+            ts = tsjson.0;
+        }
+
+        let cur_duration_secs = SystemTime::now().duration_since(UNIX_EPOCH)?;
+
+        let mut node_status = NodeStatusInfo::new(cur_duration_secs, tipsets)?;
+        node_status.set_network(&network);
+        node_status.set_node_start_time(start_time);
+        node_status.set_default_wallet(default_wallet_address);
+
+        let status_output = node_status.display(&opts.color);
+
+        println!("Network: {}", status_output.network);
+        println!("Uptime: {}", status_output.uptime);
+        println!("Chain: {}", status_output.chain_status);
+        println!("Chain health: {}", status_output.health);
+        println!("Default wallet address: {}", status_output.wallet_address);
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{str::FromStr, sync::Arc, time::Duration};
+
+    use chrono::Local;
+    use colored::*;
+    use forest_blocks::{BlockHeader, Tipset};
+    use forest_cli_shared::logger::LoggingColor;
+    use forest_shim::{address::Address, econ::TokenAmount};
+    use fvm_shared::clock::EPOCH_DURATION_SECONDS;
+    use quickcheck_macros::quickcheck;
+
+    use super::NodeStatusInfo;
+    use crate::cli::info_cmd::SyncStatus;
+
+    fn mock_tipset_at(seconds_since_unix_epoch: u64) -> Arc<Tipset> {
+        let mock_header = BlockHeader::builder()
+            .miner_address(Address::from_str("f2kmbjvz7vagl2z6pfrbjoggrkjofxspp7cqtw2zy").unwrap())
+            .timestamp(seconds_since_unix_epoch)
+            .build()
+            .unwrap();
+        let tipset = Tipset::from(&mock_header);
+
+        Arc::new(tipset)
+    }
+
+    fn node_status_good() -> NodeStatusInfo {
+        super::NodeStatusInfo {
+            behind: Duration::from_secs(0),
+            health: 90.,
+            epoch: i64::MAX,
+            base_fee: TokenAmount::from_whole(1),
+            sync_status: SyncStatus::Ok,
+            start_time: Local::now(),
```

Comment:
It would be great if you didn't use non-deterministic code in the tests.

Reply:
Using `DateTime::<Utc>::MIN_UTC`